.github/workflows/tests.yml

```diff
@@ -13,57 +13,95 @@ jobs:
         with:
           python-version: 3.9
       - name: Install packages
-        run: sudo apt-get install tidy environment-modules -y; pip install cryptography;
+        run: pip install -U pip; pip install cryptography; pip install jupyter_client==6.1.12; pip install ipython_genutils==0.2.0; pip install nbconvert==5.6.1; pip install rpy2; pip install ipykernel; pip install pandas==1.2.3; pip install pillow; pip install -U Jinja2==3.0
```

Comment:
this shouldn't be here, if the jupyter env isn't working we should merge in my branch

Reply:
Which part specifically shouldn't be here? The jupyter env is working